kvserver/rangefeed: BenchmarkRangefeed panic

To reproduce:
```
./dev bench ./pkg/kv/kvserver/rangefeed --timeout=20m --filter=BenchmarkRangefeed --count=1 --bench-time=1s --bench-mem -v --stream-output --ignore-cache
```

Output:

```
$ bazel test --test_timeout=1200 pkg/kv/kvserver/rangefeed:all --nocache_test_results --test_arg -test.run=- --test_arg -test.bench=BenchmarkRangefeed --test_sharding_strategy=disabled --test_arg -test.v --test_arg -test.benchtime=1s --test_arg -test.benchmem --crdb_test_off --test_env COCKROACH_TEST_FIXTURES_DIR=/Users/herko/Library/Caches/crdb-test-fixtures --sandbox_writable_path=/Users/herko/Library/Caches/crdb-test-fixtures --test_output streamed
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 415a7bb8-d8ed-46c0-963b-efc348ece4aa
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Analyzed 2 targets (845 packages loaded, 25402 targets configured).
INFO: Found 1 target and 1 test target...
goos: darwin
goarch: arm64
BenchmarkRangefeed
BenchmarkRangefeed/procType=legacy/opType=write/numRegs=1
BenchmarkRangefeed/procType=legacy/opType=write/numRegs=1-10         	 1467762	       778.1 ns/op	     429 B/op	       7 allocs/op
BenchmarkRangefeed/procType=legacy/opType=write/numRegs=10
BenchmarkRangefeed/procType=legacy/opType=write/numRegs=10-10        	  227383	      5440 ns/op	     512 B/op	       8 allocs/op
BenchmarkRangefeed/procType=legacy/opType=write/numRegs=100
BenchmarkRangefeed/procType=legacy/opType=write/numRegs=100-10       	   20037	     85438 ns/op	     514 B/op	       9 allocs/op
BenchmarkRangefeed/procType=legacy/opType=commit/numRegs=1
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x10629f904]

goroutine 685 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x1400204c060?, {0x106f15200, 0x1401c3df080})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x68
panic({0x106bbf8c0?, 0x107ffd8d0?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*txnPushAttempt).pushOldTxns(0x140005e8f30, {0x106f15200, 0x1401c3df080})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed/task.go:266 +0x34
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*txnPushAttempt).Run(0x1400050b718?, {0x106f15200, 0x1401c3df080})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed/task.go:254 +0x54
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x128
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 654
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x31c
I240331 10:53:16.122162 1 (gostd) testmain.go:207  [-] 1  Test //pkg/kv/kvserver/rangefeed:rangefeed_test exited with error code 2
```

--- a/pkg/kv/kvserver/rangefeed/processor.go
+++ b/pkg/kv/kvserver/rangefeed/processor.go
@@ -122,11 +122,22 @@ type Config struct {
 // SetDefaults initializes unset fields in Config to values
 // suitable for use by a Processor.
 func (sc *Config) SetDefaults() {
-	if sc.PushTxnsInterval == 0 {
-		sc.PushTxnsInterval = DefaultPushTxnsInterval
-	}
-	if sc.PushTxnsAge == 0 {
-		sc.PushTxnsAge = defaultPushTxnsAge
+	// Some tests don't set the TxnPusher, so we avoid setting a default push txn
+	// interval in such cases #121429.
+	if sc.TxnPusher == nil {
+		if sc.PushTxnsInterval != 0 {
+			panic("nil TxnPusher with non-zero PushTxnsInterval")
+		}
+		if sc.PushTxnsAge != 0 {
+			panic("nil TxnPusher with non-zero PushTxnsAge")
+		}
+	} else {
+		if sc.PushTxnsInterval == 0 {
+			sc.PushTxnsInterval = DefaultPushTxnsInterval
+		}
+		if sc.PushTxnsAge == 0 {
+			sc.PushTxnsAge = defaultPushTxnsAge
+		}
 	}
 }
 

